app/lib/account/default_auth_provider.dart

```diff
@@ -86,6 +88,72 @@ class DefaultAuthProvider extends BaseAuthProvider {
     final openIdData = await openIdDataFetch();
     return await token.verifySignature(openIdData.jwks);
   }
+
+  String _getOauthSiteAudience() => activeConfiguration.pubSiteAudience!;
+
+  Future<String?> _getOauthSiteAudienceSecret(String audience) async {
+    return await secretBackend
+        .getCachedValue('${SecretKey.oauthPrefix}$audience');
+  }
+
+  @override
+  Future<Uri> getOauthAuthenticationUrl({
+    required String state,
+    required String nonce,
+  }) async {
+    // Using https://developers.google.com/identity/protocols/oauth2/web-server#httprest_1
+    return Uri.parse('https://accounts.google.com/o/oauth2/v2/auth').replace(
+      queryParameters: {
+        'client_id': _getOauthSiteAudience(),
+        'redirect_uri': getOauthCallbackUrl(),
+        'response_type': 'code',
+        'scope': [
+          oauth2_v2.Oauth2Api.openidScope,
+          oauth2_v2.Oauth2Api.userinfoEmailScope,
+          oauth2_v2.Oauth2Api.userinfoProfileScope,
+        ].join(' '),
+        'state': state,
+        'nonce': nonce,
+      },
+    );
+  }
+
+  @override
+  Future<AuthResult?> tryAuthenticateOauthCode({
+    required String code,
+    required String expectedNonce,
+  }) async {
+    try {
```

Comment:
Don't we also need to verify the `state`:
https://developers.google.com/identity/openid-connect/openid-connect#confirmxsrftoken